buổi-10/main.js

```diff
@@ -0,0 +1,150 @@
+   }
+}
+
+// bài 3
+const getSum = number => {
+    let sum=0
+    if(number >= 1 && number <= 30) {
+        for(let i = 0; i<= number; i++) {
+            sum+=i
+        }
+        return sum
+    } else {
+        console.log('Không thỏa mãn điều kiện của n!!')
+    }
+}
+
+// bài 4
+const factorial = number => {
+    if(number === 0){
+        return 1
+    }
+    return number * factorial(number - 1)
+}
+const Result = number => {
+    if(number >= 1 && number <= 30) {
+       return factorial(number)
+    } else {
+       console.log('Không thỏa mãn điều kiện của n!!')
+    }
+}
+
+// bài 5
+const number = [1, 2, 3, 6, 2, 9, 10]
+const getEvenNumber= (number=[]) => {  
+    let count=0
+    for (let i = 0; i < number.length; i++) {  
+        if(number[i]% 2 === 0) {
+            count+=1
+        }
+    }
+    return count;
+}
+
+// bài 6
+arr = ['A', 'C', 'A', 'A', 'B', 'D', 'B']
+const deleteElement = arr => {
+    let newArr = []
+    for (let i = 0; i < arr.length ; i++) {
+        if (newArray.indexOf(arr[i]) === -1) {
+            newArray.push(arr[i])
+        }
+    } 
+    return newArr
+}
+
+// bài 7
+const studentNames = [
+    { id: 1, name: 'Nguyễn Thị Tèo' },
+    { id: 2, name: 'Phạm Văn Bưởi' },
+    { id: 3, name: 'Hoàng Văn Nam' },
+    { id: 4, name: 'Vũ Ngọc Duy' },
+    { id: 5, name: 'Nguyễn Minh Nhật' },
+    { id: 6, name: 'Phí Duy Quân' },
+    { id: 7, name: 'Trần Minh Minh' }
+]
+const studentScores = [
+    { id: 1, score: 9.2 },
+    { id: 2, score: 2.3 },
+    { id: 3, score: 3.7 },
+    { id: 4, score: 6.9 },
+    { id: 5, score: 5.2 },
+    { id: 6, score: 9.6 },
+    { id: 7, score: 6.1 }
+]
+const joinStudents= (studentNames, studentScores) => {
+    for (let i = 0; i < studentNames.length; i++) {
+        for (let j = 0; j < studentScores.length; j++) {
+            if (studentNames[i].id === studentScores[j].id) {
+                studentNames[i].score = studentScores[j].score
+            }
+        }
+    }
+    return studentNames
+}
+
+// bài 8
+const students = [
+    { id: 1, name: 'Nguyễn Thị Tèo', score: 9.2 },
+    { id: 2, name: 'Phạm Văn Bưởi', score: 2.3 },
+    { id: 3, name: 'Hoàng Văn Nam', score: 3.7 },
+    { id: 4, name: 'Vũ Ngọc Duy', score: 6.9 },
+    { id: 5, name: 'Nguyễn Minh Nhật', score: 5.2 },
+    { id: 6, name: 'Phí Duy Quân', score: 9.6 },
+    { id: 7, name: 'Trần Minh Minh', score: 6.1 }
+]
+const maxScore = students => {
+    let maxScore = students[0].score
+    for(let i = 1; i < students.length; i++) {
+        if (maxScore < students[i].score) {
+            maxScore = students[i].score
+        }
+    }
+    return maxScore
+}
+const minScore = students => {
+    let minScore = students[0].score
+    for(let i = 1; i < students.length; i++) {
+        if (minScore < students[i].score) {
+            minScore = students[i].score
+        }
+    }
+    return minScore
+}
+const findMinAndMaxStudentsScore = students => {
+    let theBest = []
+    let theBad = []
```

Comment:
2 biến này bên dưới ko bị gán lại
do vậy hãy đặt là `const`

```suggestion
    const theBest = []
    const theBad = []
```